src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -15,6 +15,8 @@
 
     List<Letter> findFirst3ByStatus(LetterStatus status);
 
+    List<Letter> findFirst3ByStatusAndEncryptionKeyFingerprint(LetterStatus status, String fingerprint);
```

Comment:
IMO the method name can be shorted

Reply:
note that the SQL query is autogenerated by JPA here, it's based on the method name.